File: seoyeon/week2/다트게임.py
def solution(dartResult):
    answer = 0
    spc = ''
    # 분리작업
    for i in dartResult :
        if i != '#' and i != '*' and i != 'S' and i != 'D' and i != 'T' :
            spc += i
        else : 
            spc += i
            spc += ' '
    tmp = spc.split(' ')
    tmp = tmp[:-1]
    # S, D, T 적용
    for i in range(len(tmp)) :
        if tmp[i][-1] == 'S' : 
            tmp[i] = int(tmp[i][:-1]) ** 1
        elif tmp[i][-1] == 'D' :
            tmp[i] = int(tmp[i][:-1]) ** 2
        elif tmp[i][-1] == 'T' :
            tmp[i] = int(tmp[i][:-1]) ** 3
    # *과 #처리
    for i in range(len(tmp)) :
        if tmp[i] == '*' :
            if tmp.index(tmp[i]) == 1 :
                tmp[i-1] *= 2
            else :
                if tmp[i-2] != '#' and tmp[i-2] != '*' and tmp[i-2] != ' ':
                    tmp[i-2] *= 2
                else :
                    tmp[i-3] *= 2
                tmp[i-1] *= 2
            tmp[i] = ' '
        elif tmp[i] == '#' :
            tmp[i-1] *= -1
            tmp[i] = ' '
    # 합 구하기
    for i in tmp :
        if type(i) == int :
            answer += i
    return answer

File: yejee/week2/다트 게임.java
import java.util.Arrays;
class Solution {
    public int solution(String dartResult) {
        int answer = 0;
        int[] score = {0, 0, 0};
        int n = 0; // score의 index용
        
        for (int i = 0; i < dartResult.length(); i++) {
            char ch = dartResult.charAt(i);
            if ('0' <= ch && ch <= '9') {
                score[n] = ch - '0';
                if(dartResult.charAt(i + 1) == '0'){
                    score[n] *= 10;
                    i++;
                }
            }
            else if(ch == 'S' || ch == 'D' || ch == 'T'){ 
                score[n] = SDT(score[n], ch); //해당 기회의 점수 계산
                if(i != dartResult.length() - 1 && '0' <= dartResult.charAt(i + 1) && dartResult.charAt(i + 1) <= '9'){ // 여기에 해당하는 경우 상을 받지 않는다는 의미이므로 score의 index를 넘겨준다.
                    n++; 
                }
            }
            else if(ch == '*'){
                score = star(score, n);
                n++;
            }
            else{
                score[n] *= -1;
                n++;
            }
        }
        
        answer = score[0] + score[1] + score[2];
        return answer;
    }
    // 싱글 더블 트리플 계산하는 함수
    public int SDT(int score, char c){
        if(c == 'S')
            return score;
        else if(c == 'D')
            return score*score;
        else 
            return score*score*score;
    }
    // 스타상 계산하는 함수
    public int[] star(int[] score, int n){
        if(n == 0){
            score[n] *= 2;
        }
        else{
            score[n-1] *= 2;
            score[n] *= 2;
        }
        return score;
    }
}